crates/cargo-contract/src/cmd/config.rs

```diff
@@ -0,0 +1,213 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of cargo-contract.
+//
+// cargo-contract is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// cargo-contract is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with cargo-contract.  If not, see <http://www.gnu.org/licenses/>.
+
+use ink_env::{
+    DefaultEnvironment,
+    Environment,
+};
+use sp_core::Pair;
+use std::{
+    fmt::Debug,
+    str::FromStr,
+};
+use subxt::{
+    config::PolkadotExtrinsicParams,
+    tx::{
+        PairSigner,
+        Signer as SignerT,
+    },
+    utils::MultiAddress,
+    Config,
+    SubstrateConfig,
+};
+
+/// Configuration for signer
+pub trait SignerConfig<C: Config + Environment> {
+    type Signer: SignerT<C> + FromStr + Clone;
+}
+
+/// A runtime configuration for the ecdsa test chain.
+/// This thing is not meant to be instantiated; it is just a collection of types.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Ecdsachain {}
+
+impl Config for Ecdsachain {
+    type Hash = <SubstrateConfig as Config>::Hash;
+    type AccountId = <SubstrateConfig as Config>::AccountId;
+    type Address = MultiAddress<Self::AccountId, ()>;
+    type Signature = <SubstrateConfig as Config>::Signature;
+    type Hasher = <SubstrateConfig as Config>::Hasher;
+    type Header = <SubstrateConfig as Config>::Header;
+    type ExtrinsicParams = PolkadotExtrinsicParams<Self>;
+    type AssetId = <SubstrateConfig as Config>::AssetId;
+}
+
+impl Environment for Ecdsachain {
+    const MAX_EVENT_TOPICS: usize = <DefaultEnvironment as Environment>::MAX_EVENT_TOPICS;
+    type AccountId = <DefaultEnvironment as Environment>::AccountId;
+    type Balance = <DefaultEnvironment as Environment>::Balance;
+    type Hash = <DefaultEnvironment as Environment>::Hash;
+    type Timestamp = <DefaultEnvironment as Environment>::Timestamp;
+    type BlockNumber = <DefaultEnvironment as Environment>::BlockNumber;
+    type ChainExtension = <DefaultEnvironment as Environment>::ChainExtension;
+}
+
+impl SignerConfig<Self> for Ecdsachain
+where
+    <Self as Config>::Signature: From<sp_core::ecdsa::Signature>,
+{
+    type Signer = SignerEcdsa<Self>;
+}
+
+/// A runtime configuration for the Polkadot based chain.
+/// This thing is not meant to be instantiated; it is just a collection of types.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum Polkadot {}
```

Comment:
```suggestion
pub enum Substrate {}
```

I'm leaning towards renaming it to Substrate, as that's more correct and includes also solochains like Aleph Zero.

Reply:
In `subxt` there is another config called [`Substrate`](https://github.com/paritytech/subxt/blob/fd0fe5700281ecd9a11d551f0aa6d4ee21ee6460/subxt/src/config/substrate.rs#L19), which matches the default kitchen sink node extrinsic params. So in `subxt` `Polkadot` -> relay chain and `node-template` config (used by most parachains and `substrate-contracts-node`), and `Substrate` -> substrate kitchen sink node with different extrinsic params.

We can give these different names possibly here though.

Reply:
Added Substrate config